File: problems/11/solution.py
class Solution:
    def oddEvenLinkedList(self, head):
        if not head:
            return None
        odd = head
        even = head.next
        evenHead = even

        while even != None and even.next != None:
            odd.next = even.next
            odd = odd.next
            even.next = odd.next
            even = even.next

        odd.next = evenHead
        return head

File: problems/11/solution.c
ListNode* oddEvenLinkedList(ListNode* head) {
	if (!head) {
		return head;
	}
	ListNode* odd = head;
	ListNode* even = head -> next;
	ListNode* evenHead = even;
	while (even != NULL && even->next != NULL) {
		odd->next = even->next;
		odd = odd->next;
		even->next = odd->next;
		even = even->next;
	}
	odd->next = evenHead;
	return head;
}
